### PR TITLE
{Compute} Change `KeyVaultClient` to `CertificateClient` in vm module

### PR DESCRIPTION
**Pull request by ReaNAiveD**
**Related command**

- `vm secret add`
- `vm encryption enable`
- `vmss encryption enable`

**Description**

Package `azure-keyvault` is incompatible with Python 3.11. Change `azure.keyvault.KeyVaultClient` to `azure.keyvault.certificates.CertificateClient` in vm module

**Testing Guide**

- `vm secret add -g myGroup -n vm --keyvault vault --certificate cert`
- `vm encryption enable -g myGroup -n vm --disk-encryption-keyvault vault`
- `vmss encryption enable -g myGroup -n vmss --disk-encryption-keyvault vault`

**History Notes**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

**Review by zhoxing-ms — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Comment by jiasli** [comment]
It will be great if the PR description can explain why `azure-keyvault` is incompatible with Python 3.11 (#26923).

**Comment by ReaNAiveD** [comment]
> It will be great if the PR description can explain why `azure-keyvault` is incompatible with Python 3.11 (#26923).

Updated

**Comment by jiasli** [comment]
The new PR description looks perfect! 🎉

File: src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py [hunk 1]
```diff
@@ -105,9 +105,9 @@ def check_existence(cli_ctx, value, resource_group, provider_namespace, resource
         return False
 
 
-def create_keyvault_data_plane_client(cli_ctx):
-    from azure.cli.command_modules.keyvault._client_factory import keyvault_data_plane_factory
-    return keyvault_data_plane_factory(cli_ctx)
+def create_keyvault_data_plane_client(cli_ctx, vault_base_url):
+    from azure.cli.command_modules.keyvault._client_factory import data_plane_azure_keyvault_certificate_client
+    return data_plane_azure_keyvault_certificate_client(cli_ctx, {"vault_base_url": vault_base_url})
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py:110]: This `create_keyvault_data_plane_client` functions returns a `azure.keyvault.certificates.CertificateClient`, making the name of this function no longer valid.

Comment (ReaNAiveD) [reply src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py:110]: > This `create_keyvault_data_plane_client` functions returns a `azure.keyvault.certificates.CertificateClient`, making the name of this function no longer valid.

I see. I will create a new PR to implement this change.

**Pull request merged**